### DISCUSSION TITLE
How to copy user/repo out of a GitHub URL?

### DISCUSSION THREAD

**Discussion by limaceous-bushwhacker** [gdh1995/vimium-c#1071] [Q&A] [answered]
For instance, if I'm on https://github.com/gdh1995/vimium-c/discussions, I want to extract just `gdh1995/vimium-c` and copy it to clipboard. Be better if the keybind was only applicable to GitHub.

**Comment by gdh1995** [comment]
Like this: https://github.com/gdh1995/vimium-c/issues/349#issuecomment-853785589

Add such a custom key mapping:
```
map ya copyCurrentUrl sed=1
```

And add such a text substitution rule:
```
1@^https://github.com/([^/]+/[^/]+)@$1@,matched
```

**Reply by limaceous-bushwhacker** [reply]
Ty. How do I make sure this works only on GitHub?

**Reply by gdh1995** [reply]
By `1@^https://github.com/([^/]+/[^/]+)@$1@,matched`

**Reply by limaceous-bushwhacker** [reply]
Seems to be working on all pages?

**Reply by gdh1995** [reply]
Can not reproduce `working on all pages`.

The regexp pattern of `^https://github.com/([^/]+/[^/]+)` should ensure it's only applied on GitHub sites.

**Reply by limaceous-bushwhacker** [reply]
![box](https://github.com/gdh1995/vimium-c/assets/67668779/36988795-0077-4580-b000-39bf5a1d562d)

**Reply by gdh1995** [reply]
Um, you may do another test on Google or Bing pages.

The snapshot is because Vimium C auto shows a URL's `path` if only it belongs to an extension - while the real copied is just the whole URL.

**Reply by limaceous-bushwhacker** [reply]
Uh, not sure. Seems to work on my end for some reason.

Here it is on Google:
![Google](https://github.com/gdh1995/vimium-c/assets/67668779/22af48d6-94b8-447e-a3bc-0492d741a0b6)

**Reply by gdh1995** [reply]
So I think it works "as expected".

**Comment by gdh1995** [comment]
>  if the keybind was only applicable to GitHub.

Do you mean a key should do nothing on other websites?

**Reply by gdh1995** [reply]
I once think you want to copy full URL on normal sites, while given parts on GitHub.

**Comment by gdh1995** [comment, marked as answer]
You may use `env` in key mappings:
```
env github host="https://github.com/"
run ya blank expect="github:copyCurrentUrl#sed=1"
```
And add such a text substitution rule:
```
1@^https://github.com/([^/]+/[^/]+)@$1@,matched
```
Then `ya` will do nothing on other sites.

Please note that `ya` will always be hooked by Vimium C, and there's no way to exclude a multi-key key binding on some websites.

If you assign a single key to this mapping, then you may use `Exclusion rules` to make most websites ignore a key, like
![image](https://github.com/gdh1995/vimium-c/assets/5547703/477a1e0e-0675-43b8-86a6-4f2e93d28c7f)

See https://github.com/gdh1995/vimium-c/wiki/Map-a-key-to-different-commands-on-different-websites

**Reply by limaceous-bushwhacker** [reply]
Ty, this works exactly how I'd like! I'm gonna go ahead and close this discussion.
